src/app/features/theme-drawer/theme-list-item.tsx

```diff
@@ -0,0 +1,33 @@
+import { useCallback } from 'react';
+import { BoxProps } from '@stacks/ui';
+
+import { useAnalytics } from '@app/common/hooks/analytics/use-analytics';
+import { getThemeLabel, UserSelectedTheme, useThemeSwitcher } from '@app/common/theme-provider';
+import { ThemeListItemLayout } from './theme-list-item-layout';
+
+interface ThemeListItemProps extends BoxProps {
+  theme: UserSelectedTheme;
+}
+export const ThemeListItem = ({ theme, ...props }: ThemeListItemProps) => {
+  const [_currentTheme, userSelectedTheme, setUserSelectedTheme] = useThemeSwitcher();
+  const isActive = theme === userSelectedTheme;
+  const analytics = useAnalytics();
+
+  const handleItemClick = useCallback(() => {
+    void analytics.track(`theme_selected_${theme}`);
+    setUserSelectedTheme(theme);
+  }, [analytics, setUserSelectedTheme, theme]);
+
+  const themeLabel = getThemeLabel(theme);
+
+  if (!themeLabel) return null;
```

Comment:
Is there ever not a theme label? Given the types, the compiler should know it's never undefined I think. Is that not the case.

Reply:
The types can probably be improved: it's because the underlying Map could have changed since inception, hence undefined could be returned when getting one of it's props. It isn't actually being changed, but the compiler doesn't know that.

I'll refactor the types, I think this will be better,

```tsx
export const themeLabelMap = {
  light: 'Light',
  dark: 'Dark',
  system: 'System',
};

export type UserSelectedTheme = keyof typeof themeLabelMap;
export type ComputedTheme = keyof Pick<typeof themeLabelMap, 'light' | 'dark'>;

export const getThemeLabel = (theme: UserSelectedTheme) => {
  return themeLabelMap[theme];
};
```